core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.directive.ts

```diff
@@ -64,6 +66,18 @@ angular.module('oppia').directive('topNavigationBar', [
           var truncateNavbarDebounced =
             DebouncerService.debounce(truncateNavbar, 500);
 
+          ctrl.CLASSROOM_PAGE_IS_SHOWN = false;
+
+          ctrl.updateShowClassroom = function() {
+            ClassroomBackendApiService.fetchClassroomData('math').then(
+              function(classroomData) {
+                ctrl.CLASSROOM_PAGE_IS_SHOWN = (
+                  classroomData.classroom_page_is_shown);
+              }, function(errorResponse) {
+                ctrl.CLASSROOM_PAGE_IS_SHOWN = false;
```

Comment:
Er, Why are we expecting to have an error as a correct thing? Can we create a separate handler for the status?

Reply:
Addressed.